app/livechat/server/api/lib/visitors.js

```diff
@@ -62,13 +63,40 @@ export async function findChatHistory({ userId, roomId, visitorId, pagination: {
 	});
 
 	const total = await cursor.count();
-
 	const history = await cursor.toArray();
-
-	return {
-		history,
-		count: history.length,
-		offset,
-		total,
-	};
+	if(text == 'null'){
```

Comment:
You don't need any `if/else` condition, you just need an `if` condition.
If the `text` variable is assigned, then you will perform the additional search of each room.

Reply:
done